src/utils/UrqlGuaranteedMemo.re

```diff
@@ -0,0 +1,15 @@
+// React.useMemo has to semantic guarantee so we define our own
+let useGuaranteedMemo1 = (computeValue, arg) => {
+  let (memoizedState, setMemoizedState) =
```

Comment:
Nice, this looks great 🚀 I'm down to include this in our `hooks` folder since I envision we may want to use this in other places. I am trying to think through the implications of this value being tracked in state. Upstream in `urql`'s `useSubscription`, subscription data is already kept in `state`, meaning we get a re-render every time a new message comes in, which would run this hook. I'm wondering if the `state` update here will then trigger _another_ re-render. I'm assuming React would catch them, but wondering if we may want to use a `ref` here instead. Let's keep it as is for now, and I'll do some experiments to see if we're getting additional unnecessary renders.

Reply:
Please do! I did not consider all that carefully if this is the best approach or if there are patterns already established. I also initially considered using a ref, but didn't immediately see a way to make it work since useEffect is async and you need a way to trigger a re-render after it runs.

Reply:
Actually, now that I am thinking about it, perhaps `useCallback` is the better solution? I don't see any restrictions about the semantics of `useCallback` in the React documentation.
```
  let memoizedUrqlResponseToReason = React.useCallback1(
    () => urqlResponseToReason(jsResponse),
    [|jsResponse|]
  );

  memoizedUrqlResponseToReason()
```